src/SFA.DAS.AssessorService.Application/Handlers/Certificates/UpdateCertificatesPrintStatusHandler.cs

```diff
@@ -0,0 +1,79 @@
+    {
+        private readonly ICertificateRepository _certificateRepository;
+        private readonly IMediator _mediator;
+        private readonly ILogger<UpdateCertificatesPrintStatusHandler> _logger;
+
+        public UpdateCertificatesPrintStatusHandler(ICertificateRepository certificateRepository, IMediator mediator, ILogger<UpdateCertificatesPrintStatusHandler> logger)
+        {
+            _certificateRepository = certificateRepository;
+            _mediator = mediator;
+            _logger = logger;
+        }
+
+        public async Task<ValidationResponse> Handle(UpdateCertificatesPrintStatusRequest request, CancellationToken cancellationToken)
+        {
+            var validationResult = new ValidationResponse();
+
+            var notFoundBatches = new List<int>();
+            var validPrintStatus = new List<string> { CertificateStatus.Printed, CertificateStatus.Delivered, CertificateStatus.NotDelivered };
+            
+            foreach (var certificatePrintStatus in request.CertificatePrintStatuses)
+            {
+                if (!notFoundBatches.Contains(certificatePrintStatus.BatchNumber))
+                {
+                    if(await _mediator.Send(new GetForBatchNumberBatchLogRequest { BatchNumber = certificatePrintStatus.BatchNumber }) == null)
+                    {
+                        validationResult.Errors.Add(new ValidationErrorDetail(nameof(request.CertificatePrintStatuses), $"The batch number {certificatePrintStatus.BatchNumber} was not found.", ValidationStatusCode.NotFound));
+                        notFoundBatches.Add(certificatePrintStatus.BatchNumber);
+                    }
+                }
+
+                if (!notFoundBatches.Contains(certificatePrintStatus.BatchNumber))
+                {
+                    var certificate = await _certificateRepository.GetCertificate(certificatePrintStatus.CertificateReference);
+                    if (certificate == null)
+                    {
+                        validationResult.Errors.Add(new ValidationErrorDetail(nameof(request.CertificatePrintStatuses), $"The certificate reference {certificatePrintStatus.CertificateReference} was not found.", ValidationStatusCode.NotFound));
+                    }
+                    else
+                    {
+                        // when the certificate batch number is not set then a reprint has been requested but not sent to printer
+                        // any print status update would be for a prior batch number and would not update the certificate status
+                        var changesCertificateStatus = certificatePrintStatus.BatchNumber >= (certificate.BatchNumber ?? int.MaxValue) &&
+                            certificatePrintStatus.StatusChangedAt > certificate.LatestChange().Value &&
+                            certificate.Status != CertificateStatus.Deleted;
+
+                        if (!validPrintStatus.Contains(certificatePrintStatus.Status))
```

Comment:
Could this check be done first thing within this loop to shortcut processing for an invalid status. Would save multiple roundtrips to the DB